metis-normalization/src/main/java/eu/europeana/normalization/dates/extraction/extractors/EdtfDateExtractor.java

```diff
@@ -0,0 +1,97 @@
+package eu.europeana.normalization.dates.extraction.extractors;
+
+import static eu.europeana.normalization.dates.edtf.DateQualification.CHECK_QUALIFICATION_PATTERN;
+import static eu.europeana.normalization.dates.edtf.InstantEdtfDateBuilder.OVER_4_DIGITS_YEAR_PREFIX;
+import static eu.europeana.normalization.dates.edtf.InstantEdtfDateBuilder.THRESHOLD_4_DIGITS_YEAR;
+
+import eu.europeana.normalization.dates.DateNormalizationExtractorMatchId;
+import eu.europeana.normalization.dates.DateNormalizationResult;
+import eu.europeana.normalization.dates.edtf.DateQualification;
+import eu.europeana.normalization.dates.edtf.InstantEdtfDate;
+import eu.europeana.normalization.dates.edtf.InstantEdtfDateBuilder;
+import eu.europeana.normalization.dates.edtf.Iso8601Parser;
+import eu.europeana.normalization.dates.extraction.DateExtractionException;
+import java.lang.invoke.MethodHandles;
+import java.time.temporal.TemporalAccessor;
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The pattern for EDTF dates and compatible with ISO 8601 dates.
+ * <p>This parser supports partial Level0 and Level1 from the <a href="https://www.loc.gov/standards/datetime/">Extended
+ * Date/Time Format (EDTF) Specification</a>. It only validates the date part of a date and the time if existent is discarded.
+ * Specifically from Level1, seasons and unspecified digit(s) from the right are not supported
+ * </p>
+ */
+public class EdtfDateExtractor extends AbstractDateExtractor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private static final Iso8601Parser ISO_8601_PARSER = new Iso8601Parser();
+
+  @Override
+  public DateNormalizationResult extract(String inputValue,
+      boolean flexibleDateBuild) throws DateExtractionException {
+    final InstantEdtfDate instantEdtfDate = extractInstant(inputValue, flexibleDateBuild);
+    return new DateNormalizationResult(DateNormalizationExtractorMatchId.EDTF, inputValue, instantEdtfDate);
+  }
+
+  private InstantEdtfDate extractInstant(String dateInput, boolean flexibleDateBuild) throws DateExtractionException {
+    final InstantEdtfDate instantEdtfDate;
+    Integer longYear;
```

Comment:
take into consideration to replace `longYear` to `fourDigitsYear` to avoid ambiguous interpretation with java `long` type for methods and variables in this particular class.

Reply:
I've updated similar to the others.